File: 1160.find-words-that-can-be-formed-by-characters/solution.kt
class Solution {
    fun countCharacters(words: Array<String>, chars: String): Int {
        val freq = ('a'..'z').associate { ch -> ch to chars.count { it == ch } }
        
        fun good(word: String) =
            freq.all { (ch, num) -> num >= word.count { it == ch } }
 
        return words
            .filter { good(it) }
            .map { it.length }
            .sum()
    }
}

File: 1160.find-words-that-can-be-formed-by-characters/solution.go
func countCharacters(words []string, chars string) int {
    f := make([]int, 26)
    for _, c := range chars {
        f[c - 'a']++
    }
    
    h := make([]int, 26)
    res := 0
    
    outer:
    for _, w := range words {
        copy(h, f)
        for _, c := range w {
            k := c - 'a'
            if h[k] == 0 {
                continue outer
            }
            h[k]--
        }
        res += len(w)
    }
    
    return res
}